pkg/testing/integration/program.go

```diff
@@ -744,6 +760,16 @@ func newProgramTester(t *testing.T, opts *ProgramTestOptions) *ProgramTester {
 	}
 }
 
+// NewBackendURL creates a temporary backend directory which will clean up on test exit.
+func NewBackendURL(t *testing.T) string {
```

Comment:
Can we name this better? I look at the name and it gives me no clue this opts me into using a filestate backend in a tempdir.
 
`NewTempFilestateBackendURL`? 

A bit verbose.

Reply:
`MakeTempBackend`, which is reminiscent of `MkDirTemp`?